platform-coop-toolkit/coop-library
# feat: settings page (resolves #128)

## Description

This PR adds a settings page and gives users the ability to opt-in (via a cookie) to having their recently viewed resources tracked via Koko Analytics. If the `track_viewed_resources` cookie is set to `on`, the [`koko_analytics_load_tracking_script` filter](https://github.com/ibericode/koko-analytics/blob/6119cd5a789b3d6d4f24d2e6640d2724fe450b47/readme.txt#L108) is set to `true`; under all other circumstances it is set to false.

## Steps to test

1. Create a page called 'Settings' and set its page template to 'Settings'.
2. Visit this page, change the configuration to 'Allow', save.

**Expected behavior:** The `track_viewed_resources` cookie is set to true and subsequent visits to the settings page reflect this. When the `track_viewed_resources` cookie is set to true, subsequently browsed resources are added to the `_koko_analytics_pages_viewed` cookie.

## Additional information

Not applicable.

## Related issues

- Resolves #128.
